pymtl3/passes/backends/verilog/test/TranslationImport_adhoc_gc_test.py

```diff
@@ -0,0 +1,97 @@
+#=========================================================================
+# TranslationImport_adhoc_test.py
+#=========================================================================
+# Author : Peitian Pan
+# Date   : Jun 5, 2019
+"""Test ad-hoc components with SystemVerilog translation and import."""
+
+import gc
+import resource
+
+import pytest
+
+from pymtl3 import *
+from pymtl3.passes.backends.verilog import VerilogPlaceholderPass
+from pymtl3.passes.rtlir.util.test_utility import get_parameter
+from pymtl3.stdlib.test import TestVectorSimulator
+
+from .. import TranslationImportPass
+
+
+# args: [attr, Bits, idx]
+def _set( *args ):
+  local_dict = {}
+  assert len(args) % 3 == 0
+  _tv_in_str = 'def tv_in( m, tv ):  \n'
+  if len(args) == 0:
+    _tv_in_str += '  pass'
+  for attr, Bits, idx in zip( args[0::3], args[1::3], args[2::3] ):
+    if isinstance(Bits, str):
+      _tv_in_str += f'  m.{attr} = {Bits}( tv[{idx}] )\n'
+    else:
+      _tv_in_str += f'  m.{attr} = {Bits.__name__}( tv[{idx}] )\n'
+  exec( _tv_in_str, globals(), local_dict )
+  return local_dict['tv_in']
+
+# args: [attr, Bits, idx]
+def _check( *args ):
+  local_dict = {}
+  assert len(args) % 3 == 0
+  _tv_out_str = 'def tv_out( m, tv ):  \n'
+  if len(args) == 0:
+    _tv_out_str += '  pass'
+  for attr, Bits, idx in zip( args[0::3], args[1::3], args[2::3] ):
+    if isinstance(Bits, str):
+      _tv_out_str += f'  assert m.{attr} == {Bits}( tv[{idx}] )\n'
+    else:
+      _tv_out_str += f'  assert m.{attr} == {Bits.__name__}( tv[{idx}] )\n'
+  exec( _tv_out_str, globals(), local_dict )
+  return local_dict['tv_out']
+
+class CasePlusOneComp:
```

Comment:
Is there a reason why this test case and `_set``_check` are not in `verilog/testcases/`?

Reply:
oh I totally forgot about this test. I will refactor the code

Reply:
fixed